### PR TITLE
Enhancement of hccl port config resolution.

### PR DESCRIPTION
**Pull request by BalaBalaYi**
### What changes were proposed in this pull request?

1. Set default port start from 64000.
2. Add logging for conflict port for the convenience of follow-up investigation.
3. Skip the processing if 'HCCL_IF_BASE_PORT' already configured for compatibility.

### Why are the changes needed?

For enhancement.

### Does this PR introduce any user-facing change?

No.

### How was this patch tested?

UT.

### CODE REVIEW

File: dlrover/python/common/constants.py [hunk 1]
```diff
@@ -307,3 +307,4 @@ class Accelerators(object):
 class AscendConstants(object):
     # By default there are 16 npu on one machine
     NPU_PER_NODE = 16
+    HCCL_PORT_START = "HCCL_IF_BASE_PORT"
```
Comment (linkerzhang) [inline dlrover/python/common/constants.py:310]: Add comments please.

File: dlrover/python/common/grpc.py [hunk 2]
```diff
@@ -104,7 +104,7 @@ def find_free_port_in_set(ports):
     raise RuntimeError(f"Fail to find a free port in {ports}")
 
 
-def find_free_port_for_hccl(start=60000) -> int:
+def find_free_port_for_hccl(start=64000) -> int:
```
Comment (linkerzhang) [inline dlrover/python/common/grpc.py:107]: move the integer value to constants?

File: dlrover/python/common/grpc.py [hunk 3]
```diff
@@ -120,6 +120,7 @@ def find_free_port_for_hccl(start=60000) -> int:
             logger.info(f"Find available port start from: {cur_start}")
             break
         except OSError:
+            logger.warning(f"Target port is already used: {checking_port}.")
```
Comment (linkerzhang) [inline dlrover/python/common/grpc.py:123]: has already been used

File: dlrover/python/elastic_agent/torch/training.py [hunk 4]
```diff
@@ -140,7 +141,7 @@ class ElasticLaunchConfig(LaunchConfig):
     network_check: bool = False
     comm_perf_test: bool = False
     node_unit: int = 1
-    training_port: int = 60000
+    training_port: int = 64000
```
Comment (linkerzhang) [inline dlrover/python/elastic_agent/torch/training.py:144]: reuse the same constant after fixing the "moving" comments above.

File: dlrover/python/tests/test_elastic_training_agent.py [hunk 5]
```diff
@@ -363,7 +364,7 @@ def test_sync_node_port(self):
             log_dir=self.config.log_dir,
         )
         agent.sync_training_ports()
-        self.assertEqual(os.environ["HCCL_IF_BASE_PORT"], "60000")
+        self.assertEqual(os.environ[AscendConstants.HCCL_PORT_START], "64000")
```
Comment (linkerzhang) [inline dlrover/python/tests/test_elastic_training_agent.py:367]: reuse the same constant as comments above.

File: dlrover/python/common/constants.py (hunk 1 above)
Comment (BalaBalaYi) [reply dlrover/python/common/constants.py:310]: done

File: dlrover/python/elastic_agent/torch/training.py (hunk 4 above)
Comment (BalaBalaYi) [reply dlrover/python/elastic_agent/torch/training.py:144]: lgtm

File: dlrover/python/tests/test_elastic_training_agent.py (hunk 5 above)
Comment (BalaBalaYi) [reply dlrover/python/tests/test_elastic_training_agent.py]: done

**Review by samplise — APPROVED** [review]
lgtm

**Pull request merged**